resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/security/YARNConfigurableCredentialManager.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.yarn.security
+
+import java.util.ServiceLoader
+
+import scala.collection.JavaConverters._
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.FileSystem
+import org.apache.hadoop.security.Credentials
+
+import org.apache.spark.SparkConf
+import org.apache.spark.deploy.security.ConfigurableCredentialManager
+import org.apache.spark.internal.Logging
+import org.apache.spark.util.Utils
+
+
+/**
+ * This class loads credential providers registered under the YARN-specific
+ * [[ServiceCredentialProvider]] interface, as well as the builtin credential providers defined
+ * in [[ConfigurableCredentialManager]].
+ */
+private[yarn] class YARNConfigurableCredentialManager(
+    sparkConf: SparkConf,
+    hadoopConf: Configuration,
+    fileSystems: Set[FileSystem]) extends Logging {
+
+  private val configurableCredentialManager =
+    new ConfigurableCredentialManager(sparkConf, hadoopConf, fileSystems)
+
+  // public for testing
+  val credentialProviders = getCredentialProviders
+
+  def obtainYARNCredentials(
+    hadoopConf: Configuration,
+    creds: Credentials): Long = {
+
+    val superInterval = configurableCredentialManager.obtainCredentials(
+      hadoopConf,
+      creds)
+
+    credentialProviders.values.flatMap { provider =>
+      if (provider.credentialsRequired(hadoopConf)) {
+        provider.obtainCredentials(hadoopConf, sparkConf, creds)
+      } else {
+        logDebug(s"Service ${provider.serviceName} does not require a token." +
+          s" Check your configuration to see if security is disabled or not.")
+        None
+      }
+    }.foldLeft(superInterval)(math.min)
+  }
+
+  private def getCredentialProviders:
+    Map[String, ServiceCredentialProvider] = {
+    val providers = loadCredentialProviders
+
+    providers.
+      filter { p => configurableCredentialManager.isServiceEnabled(p.serviceName) }
```

Comment:
Because it's weird. Why are you changing the config names if that particular feature (extensions loaded by YARN code) are not really a feature in core?

It would be just unnecessary log noise. When/if this is exposed as an actual API in core, then it would make more sense to warn about deprecating those configs.

Anyway, not a big deal.